packages/guideline-blocks-settings/src/components/BlockItemWrapper/Toolbar/Toolbar.tsx

```diff
@@ -18,19 +18,21 @@ export const Toolbar = ({ items, attachments }: ToolbarProps) => (
                 <AttachmentsToolbarButton />
             </ToolbarSegment>
         )}
-        <ToolbarSegment>
-            {items.map((item) => {
-                if (item.type === 'dragHandle') {
-                    return <DragHandleToolbarButton key={item.tooltip + item.type} {...item} />;
-                }
-                if (item.type === 'menu') {
-                    return <MenuToolbarButton key={item.tooltip + item.type} {...item} />;
-                }
-                if (item.type === 'flyout') {
-                    return <FlyoutToolbarButton key={item.tooltip + item.type} {...item} />;
-                }
-                return <ToolbarButton key={item.tooltip + item.type} {...item} />;
-            })}
-        </ToolbarSegment>
+        {items.length > 0 && (
+            <ToolbarSegment>
+                {items.map((item) => {
+                    if (item.type === 'dragHandle') {
+                        return <DragHandleToolbarButton key={item.tooltip + item.type} {...item} />;
+                    }
+                    if (item.type === 'menu') {
+                        return <MenuToolbarButton key={item.tooltip + item.type} {...item} />;
+                    }
+                    if (item.type === 'flyout') {
+                        return <FlyoutToolbarButton key={item.tooltip + item.type} {...item} />;
+                    }
+                    return <ToolbarButton key={item.tooltip + item.type} {...item} />;
```

Comment:
```suggestion
                   switch (item.type) {
                        case 'dragHandle':
                            return <DragHandleToolbarButton key={item.tooltip + item.type} {...item} />;
                        case 'menu':
                            return <MenuToolbarButton key={item.tooltip + item.type} {...item} />;
                        case 'flyout':
                            return <FlyoutToolbarButton key={item.tooltip + item.type} {...item} />;
                        default:
                            return <ToolbarButton key={item.tooltip + item.type} {...item} />;
                    }
```

Would prefer it as switch case

Reply:
Or maybe even better a toolbarButtonMap:

```typescript
const toolbarButtonMap = {
  dragHandle: DragHandleToolbarButton,
  menu: MenuToolbarButton,
  flyout: FlyoutToolbarButton,
};
``` 

and then we could do 

```typescript
{items.map((item) => {
        const ToolbarButton = toolbarButtonMap[item.type] || ToolbarButton;
        return <ToolbarButton key={`${item.tooltip}-${item.type}`} {...item} />;
})}
```

But not sure about this one

Reply:
Adjusted to the switch case version. The object approach resulted in too many type errors and fixing them would cause some overhead.